### ISSUE TITLE
Add autochdir option

### ISSUE THREAD

**Issue by smolck** [helix-editor/helix#2056]
#### Describe your feature request

In Neovim, there's an option `'autochdir'` that does the following:
```
'autochdir' 'acd'	boolean (default off)
			global
	When on, Vim will change the current working directory whenever you
	open a file, switch buffers, delete a buffer or open/close a window.
	It will change to the directory containing the file which was opened
	or selected.
	Note: When this option is on some plugins may not work.
```
I was wondering if something like this could be added to Helix?

(FWIW I actually managed [to get it somewhat working](https://github.com/smolck/helix/commit/180d8182f92cd19e3c2bdcf98f39d11ed9a4b576) but then realized what I had doesn't work for switching between splits, and probably several other edgecases I haven't considered, so figure it's probably best to make this issue about it.)

I will say that I've been using this option in Neovim for a while, but the more I think about it, this might not actually be a feature worth adding (and I might drop it from my neovim config actually). I'm honestly not sure, so interested to hear other's thoughts on this.

**Comment by the-mikedavis** [comment]
This could be interesting with something like the recently merged #1600. OTOH it seems like if I used it, I would end up having to think about which buffer I'm in before doing anything that takes the cwd into account which could be confusing.

**Comment by cjohansson** [comment]
Yes I would like this feature as well, Emacs works like this and you can have multiple projects open at the same time with different working directories

**Comment by weakphish** [comment]
@the-mikedavis

**Comment by nour-dev** [comment]
I think this is a good feature . could the code related to this config be merged to allow the intended behavior ?

**Comment by uncomfyhalomacro** [comment]
> This could be interesting with something like the recently merged #1600. OTOH it seems like if I used it, I would end up having to think about which buffer I'm in before doing anything that takes the cwd into account which could be confusing.

Maybe if it is enabled, there is an option to show the directory of the buffer. So here is a possible config options for this feature:

Defaults:

```toml
[editor]
autochdir = false  # by default, it is false
```

If enabled:

```toml
[editor.autochdir]
show-directory = false
truncation-length = 2  # truncate the directory paths as desired
```

I want it to be disabled by default since some people do get confused when it is enabled by default

**Comment by uncomfyhalomacro** [comment]
There might be also an issue with the `file-picker` though. Maybe we could add an option to the `file-picker` too as well if we want it to respect the `autochdir` option or not from the `editor` section.
